components/use-messages/message-item.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import useTheme from '../styles/use-theme'
+import { Message } from './use-message'
+import X from '@zeit-ui/react-icons/x'
+import { getStyles } from './styles'
+
+export type MessageItemProps = Message & {
+  id?: string
+  destroy?: Function
+}
+const transitionDuration = 150
+
+const MessageItem: React.FC<MessageItemProps> = React.memo(
+  ({ className, id, text, delay, destroy, onClose, closeable, ...rest }) => {
+    const theme = useTheme()
+    const { color, bgColor, icon } = useMemo(() => getStyles(theme.palette, rest.color), [
+      theme.palette,
+      rest.color,
+    ])
+    let boxShadow: string = theme.expressiveness.D2
+    if (rest.color !== 'default' && !rest.shadow) {
+      boxShadow = 'none'
+    }
+    const [visible, setVisible] = useState<boolean>(false)
+    const [hide, setHide] = useState<boolean>(false)
+    const [hover, setHover] = useState<boolean>(false)
+    useEffect(() => {
+      const timer = setTimeout(() => {
+        setVisible(true)
+        clearTimeout(timer)
+      }, 10)
+      return () => clearTimeout(timer)
+    }, [])
+    useEffect(() => {
+      let timer: any = null
+      if (hover) {
+        clearTimeout(timer)
```

Comment:
timer here will always be `null`

Reply:
maybe use ref to store timer, there's similar code in auto-complete

Reply:
done